pkg/controller/rbac/definition/reconciler.go

```diff
@@ -117,10 +117,11 @@ func WithClusterRoleRenderer(rr ClusterRoleRenderer) ReconcilerOption {
 // NewReconciler returns a Reconciler of CompositeResourceDefinitions.
 func NewReconciler(mgr manager.Manager, opts ...ReconcilerOption) *Reconciler {
 	r := &Reconciler{
-		// TODO(negz): Is Updating appropriate here? Probably.
+		// NOTE(hasheddan): Patching applicator is used to avoid conflict with
+		// Kubernetes RBAC aggregation controller.
 		client: resource.ClientApplicator{
 			Client:     mgr.GetClient(),
-			Applicator: resource.NewAPIUpdatingApplicator(mgr.GetClient()),
+			Applicator: resource.NewAPIPatchingApplicator(mgr.GetClient()),
```

Comment:
If I recall correctly the patching applicator here given that it's additive; i.e. we'll never remove any field that was previously set on a Role or ClusterRole. That said, I believe the rules array will be set to our desired state (i.e. we'd delete any rules that we no longer wanted) due to the fact that we're using a JSON merge patch that always completely rewrites arrays. I'm a little wary of depending on this behaviour for security purposes; if we do need to stick with the patching applicator I think we should at least add a comment explaining what it is doing and why it's safe.

Reply:
I do have a comment above, but I actually think we can stick with updating. The second commit is what fixed the issue from what I have been able to tell